x-pack/plugins/infra/public/components/infrastructure_node_metrics_tables/shared/hooks/metrics_to_api_options.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  MetricsExplorerOptions,
+  MetricsExplorerOptionsMetric,
+} from '../../../../pages/metrics/metrics_explorer/hooks/use_metrics_explorer_options';
+
+export interface NodeMetricsExplorerOptionsMetric<Field extends string>
+  extends Omit<MetricsExplorerOptionsMetric, 'field'> {
+  field: Field;
+}
+
+export type MetricsMap<T extends string> = {
+  [field in T]: NodeMetricsExplorerOptionsMetric<field>;
+};
+
+export function metricsToApiOptions<T extends string>(metricsMap: MetricsMap<T>, groupBy: string) {
```

Comment:
Finding this logic a bit hard to read.  Maybe we can add some documentation in the code to describe these functions and a unit test?

Reply:
Absolutely, will add!